### PR TITLE
Improve sherpa.citation (fix #994 #987)

### LINKED ISSUE #994: sherpa.citation() doesn't quite work but sherpa.citation('latest') does
Thanks to `hysterical raisons` the new-to-Sherpa-4.12.2 citation function is slightly borked

```
>>> import citation
>>> sherpa.citation()
```

isn't able to pickup the data from Zenodo, because it's missing a `created` field (and possibly others, but that's where it's complaining).

The following does work:

```
>>> sherpa.citation('latest')
```

Note to self: to find

```
>>> import sherpa
>>> import logging
>>> logging.getLogger('sherpa').setLevel(logging.DEBUG)
>>> sherpa.citation('temp-file.txt')
Unable to find metadata: 'created'
```

### PR DESCRIPTION
**Pull request by DougBurke**
# Summary

Fixes the sherpa.citation() command with its default argument (issue #994) and adds release 4.12.2 to the hard-coded list of releases. A typo in a warning message was fixed (#987).

# Details

I had not realised Zenodo was doing pagination and, because I also had not set a "most-recent" order on the query the 4.12.2 response wasn't being included in the first set of results, and so wasn't being found. We now iterate through the pages (working around https://github.com/zenodo/zenodo/issues/1662) if the requested version isn't found (and the search order has been changed so that the latest version is returned first, which means the paging mechanism shouldn't be needed as long as the hard-coded list is kept up to date).

I've added developer documentation describing how to update the sherpa._get_citation_hardcoded routine to add new releases, and added the `scripts/make_zenodo_citation.py` script which will print out the needed update to stdout.

### CODE REVIEW

**Comment by DougBurke** [comment]
The tests are marked as failed because of coverage checks only, as the Travis tests don't use the --runzenodo flag

**Review by hamogu — APPROVED** [review]
Would it make sense to wait and get 4.13 in here, too? I though that @Marie-Terrell  has already branched that off. Since this PR certainly won't make it into 4.13, there is no harm in waiting a few more days until that's on zenodo and hardcode it here, too.

Of course, that could just as easily done in the follow-up PR, so I'm approving this regardless and leave it up to @DougBurke if he wants to wait or get this in now.

**Comment by DougBurke** [comment]
Let's have a race between getting this merged or getting 4.13.0 out the door so we can hard-code the Zenodo values!

**Comment by hamogu** [comment]
I think @wmclaugh is holding off merging new stuff until 4.13 is out, so it's a race this PR is guaranteed to loose!

**Comment by DougBurke** [comment]
I've added the 4.13.0 hard-coded data.

**Pull request merged**
